File: GC.Domain/Contacts/EmergencyContacts/EmergencyContact.cs
using System;

namespace GC.Domain.Contacts.EmergencyContacts
{
    public class EmergencyContact
    {
        public Guid Id { get; }
        public EmergencyContactType Type { get; }
        public String CityPhone { get; }
        public String MobilePhone { get; }

        public EmergencyContact(Guid id, EmergencyContactType type, String cityPhone, String mobilePhone)
        {
            Id = id;
            Type = type;
            CityPhone = cityPhone;
            MobilePhone = mobilePhone;
        }
    }
}

File: GC.WebSpace/ClientApp/domain/contacts/emergencyContacts/emergencyContact.ts
import { EmergencyContactType } from "./emergencyContactType";

export class EmergencyContact {
    constructor(
        public readonly id: string,
        public readonly type: EmergencyContactType,
        public readonly cityPhone: string,
        public readonly mobilePhone: string
    ) { }
}

export namespace EmergencyContact {
    export function toModel(object: any): EmergencyContact {
        return new EmergencyContact(object.id, object.type, object.cityPhone, object.mobilePhone);
    }

    export function toModels(objects: any[]): EmergencyContact[] {
        return objects.map(toModel)
    }
}